esti/s3_gateway_test.go

```diff
@@ -456,7 +405,210 @@ func TestListMultipartUploads(t *testing.T) {
 	require.Contains(t, keys, obj2)
 }
 
-func TestListMultipartUploadsUnsupported(t *testing.T) {
+func TestMultipartListUploadsOrdering(t *testing.T) {
+	RequireBlockstoreType(t, block.BlockstoreTypeS3)
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+	s3Endpoint := viper.GetString("s3_endpoint")
+	s3Client := createS3Client(s3Endpoint, t)
+
+	// Create multipart uploads with keys that should be ordered lexicographically
+	// Using keys that might have different physical addresses but need logical ordering
+	keysPrefix := "main/"
+	testKeys := []string{
+		keysPrefix + "z-last-key",   // Should be last alphabetically
+		keysPrefix + "a-first-key",  // Should be first alphabetically
+		keysPrefix + "m-middle-key", // Should be in middle
+		keysPrefix + "b-second-key", // Should be second
+	}
+
+	// Expected order after sorting
+	expectedOrder := []string{
+		"a-first-key",
+		"b-second-key",
+		"m-middle-key",
+		"z-last-key",
+	}
+
+	// Create all multipart uploads
+	var uploadIDs []string
+	for _, key := range testKeys {
+		resp := createMultipartHelper(ctx, t, s3Client, repo, key)
+		uploadIDs = append(uploadIDs, *resp.UploadId)
+	}
+
+	// List all uploads without pagination
+	output, err := s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{
+		Bucket: aws.String(repo),
+	})
+	require.NoError(t, err, "failed to list multipart uploads")
+	require.Len(t, output.Uploads, len(testKeys), "should return all uploads")
+
+	// Verify ordering - keys should be in lexicographic order
+	actualKeys := extractUploadKeys(output)
+	require.Equal(t, expectedOrder, actualKeys, "uploads should be ordered by key lexicographically")
+
+	// Test with MaxUploads that exactly matches total uploads (request completely fulfilled)
+	outputExact, err := s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{
+		Bucket:     aws.String(repo),
+		MaxUploads: aws.Int32(4), // Exactly the number of uploads we have
+	})
+	require.NoError(t, err, "failed to list with exact max uploads")
+	require.Len(t, outputExact.Uploads, 4, "should return all 4 uploads")
+
+	// IsTruncated should be nil or false when not truncated
+	if outputExact.IsTruncated != nil {
+		require.False(t, *outputExact.IsTruncated, "should not be truncated when request is completely fulfilled")
+	}
```

Comment:
Relevant to multiple places in the test code.
In this case we require nil or or pointer to false - we can require using `apiutil.Value` or `require.True(outputExact.IsTruncated == nil || !*outputExact.IsTruncated)`.

Not a blocker.